### PR TITLE
WP-10807 : keep leading zero

### PR DESCRIPTION
**Pull request by woody-feng**

### CODE REVIEW

File: tap_s3_csv/transform.py [hunk 1]
```diff
@@ -319,9 +319,9 @@ def _transform(self, data, typ, schema, path):
         elif typ == "integer":
             if isinstance(data, str):
                 data = data.replace(",", "")
-
             try:
-                return True, int(data)
+                int(data)
+                return True, str(data)
```
Comment (woody-feng) [inline tap_s3_csv/transform.py:324]: feel like we should keep the original form of the data by kept it in string, due to confirmation screen, int(a) can discard leading zero etc

Comment (pggagnon) [reply tap_s3_csv/transform.py:324]: hmmm I’m surprised this works. I would have thought the `int(data)` that you call strip the leading zeros before `str(data)` is able to give output that we want? Why not remove `int(data)` entirely? What purpose is it serving?

Comment (woody-feng) [reply tap_s3_csv/transform.py:324]: int(data) make sure we can convert data into an int, otherwise should return False

**Review by pggagnon — DISMISSED** [review]

**Review by yi-varicent — COMMENTED** [review]
1. what happens if i put in 123,456 and want to keep it a string, the code looks like it will strip comma
2. what is the diff between typ + schema here?

**Review by yi-varicent — CHANGES_REQUESTED** [review]
as discussed

**Comment by woody-feng** [comment]
> 1. what happens if i put in 123,456 and want to keep it a string, the code looks like it will strip comma
> 2. what is the diff between typ + schema here?

**Review by yi-varicent — APPROVED** [review]
looks fine to me @kchan-varicent fyi

**Review by yi-varicent — APPROVED** [review]

File: tap_s3_csv/transform.py [hunk 2]
```diff
@@ -317,20 +318,18 @@ def _transform(self, data, typ, schema, path):
...
 
         elif typ == "number":
-            if isinstance(data, str):
-                data = data.replace(",", "")
-
             try:
-                return True, float(data)
+                num = data.replace(',', '') if isinstance(data, str) else data
+                float(num)
+                return True, str(data)
```
Comment (ichidyakin) [inline tap_s3_csv/transform.py:332]: @ali-varicent , @kchan-varicent , @yi-varicent , @woody-feng ,
In the `tap-tqp` import (which involves `tap-s3-csv` and `target-s3-csv`) this causes a failure (it started failing after these changes had been merged). A JIRA ticket: https://varicent.atlassian.net/browse/WP-10970
@ali-varicent , could you follow up on this, please, while I'm away?

**Pull request merged**